.azure/scripts/wheels.yml

```diff
@@ -2,37 +2,38 @@
 steps:
 - script: |
     mkdir -p dist
-    python -m pip install --upgrade pip
+    python -m pip install --upgrade pip setuptools -r test-requirements.txt
   displayName: 'Install dependencies'
 
 - script: |
     python -m pip wheel . -w wheelhouse/
   displayName: 'Build wheel'
-  inputs:
-    PathtoPublish: 'wheelhouse'
-    ArtifactName: 'python-wheels'
-    publishLocation: 'Container'
 
 - script: |
     ls -lh wheelhouse
     cp wheelhouse/$(package_name)* dist/.
   displayName: 'Show wheelhouse'
 
 - script: |
-    python -m pip install JPype1 --no-index -f wheelhouse
+    python -m pip install jpype1 --no-index -f wheelhouse
   displayName: 'Install module'
 
 - script: |
     python setup.py test_java
-    python -m pip install -r test-requirements.txt
-  displayName: 'Install test'
+  displayName: 'Build java tests'
 
 - script: |
     ls -l
     ls lib/
   displayName: 'Check deps'
 
+- task: PublishPipelineArtifact@0
+  inputs:
+    artifactName: 'artifact_$(Agent.JobName)_$(Agent.OS)_$(python.architecture)'
+    targetPath: 'dist'
+
 - script: |
+    rm -Rf jpype
```

Comment:
Definitely one way of avoiding importing the jpype source tree. :smile: 

In the future we could look at the `-I` flag, or ``mkdir -p not-the-source; pushd not-the-source; <run_test>; popd;``

Reply:
you are right, that one is not nice! PR welcome :)

Reply:
It was late, I was tired of reading documentation after 8 hours.   I tried shifting up a directory but discovered that some of our tests have absolute pathing.   Thus I surrendered to the need to make a quick hack.   We can just make a directory, copy the whole test tree to it, then cd to it and launch the tests as an alternative.